src/dockerconfig/docker_config.go

```diff
@@ -25,47 +25,30 @@ type DockerAuth struct {
 	Password string `json:"password"`
 }
 
-func NewDockerConfig(ctx context.Context, apiReader client.Reader, dynakube dynatracev1beta1.DynaKube) (*DockerConfig, error) {
-	var pullSecret corev1.Secret
-	if err := apiReader.Get(ctx, client.ObjectKey{Name: dynakube.PullSecret(), Namespace: dynakube.Namespace}, &pullSecret); err != nil {
-		log.Info("failed to load pull secret", "dynakube", dynakube.Name)
-		return nil, err
-	}
-	dockerAuths, err := parseDockerAuthsFromSecret(&pullSecret)
-	if err != nil {
-		log.Info("failed to parse pull secret content", "dynakube", dynakube.Name)
-		return nil, err
-	}
-
+func NewDockerConfig(apiReader client.Reader, dynakube dynatracev1beta1.DynaKube) *DockerConfig {
 	dockerConfig := DockerConfig{
 		ApiReader:     apiReader,
+		Auths:         make(map[string]DockerAuth),
 		Dynakube:      &dynakube,
-		Auths:         dockerAuths,
 		SkipCertCheck: dynakube.Spec.SkipCertCheck,
 	}
-	return &dockerConfig, nil
-
+	return &dockerConfig
 }
 
-func parseDockerAuthsFromSecret(secret *corev1.Secret) (map[string]DockerAuth, error) {
-	if secret == nil {
-		return nil, fmt.Errorf("given secret is nil")
-	}
-
-	config, hasConfig := secret.Data[".dockerconfigjson"]
-	if !hasConfig {
-		return nil, fmt.Errorf("could not find any docker config in image pull secret")
-	}
-
-	var dockerConf struct {
-		Auths map[string]DockerAuth `json:"auths"`
+func (config *DockerConfig) SetupAuths(ctx context.Context) error {
+	var pullSecret corev1.Secret
+	err := config.ApiReader.Get(ctx, client.ObjectKey{Name: config.Dynakube.PullSecret(), Namespace: config.Dynakube.Namespace}, &pullSecret)
+	if err != nil {
+		log.Info("failed to load pull secret", "dynakube", config.Dynakube.Name)
+		return errors.WithStack(err)
 	}
-	err := json.Unmarshal(config, &dockerConf)
+	dockerAuths, err := parseDockerAuthsFromSecret(&pullSecret)
 	if err != nil {
-		return nil, err
+		log.Info("failed to parse pull secret content", "dynakube", config.Dynakube.Name)
+		return err
```

Comment:
```suggestion
		return errors.WithStack(err)
```

Reply:
that error comes from an internal function, `err` is fine

Reply:
makes sense